inventory.go

```diff
@@ -41,104 +76,167 @@ type InventoryAppStats struct {
+		IconURL                   string             `json:"icon_url"`
+		IconURLLarge              string             `json:"icon_url_large"`
+		Descriptions              []DescriptionsPart `json:"descriptions"`
+		Tradable                  int                `json:"tradable"`
+		Name                      string             `json:"name"`
+		NameColor                 string             `json:"name_color"`
+		Type                      string             `json:"type"`
+		MarketName                string             `json:"market_name"`
+		MarketHashName            string             `json:"market_hash_name"`
+		Commodity                 int                `json:"commodity"`
+		MarketTradableRestriction int                `json:"market_tradable_restriction"`
+		Marketable                int                `json:"marketable"`
 	}
 
 	type Response struct {
-		Success      bool            `json:"success"`
-		ErrorMsg     string          `json:"Error"`
-		MoreStart    interface{}     `json:"more_start"` // This can be a bool or a number...
-		Inventory    json.RawMessage `json:"rgInventory"`
-		Descriptions json.RawMessage `json:"rgDescriptions"`
-		/* Missing: rgCurrency  */
+		Assets              []Asset       `json:"assets"`
+		Descriptions        []Description `json:"descriptions"`
+		Success             int           `json:"success"`
+		HasMore             int           `json:"more_items"`
+		LastAssetID         string        `json:"last_assetid"`
+		TotalInventoryCount int           `json:"total_inventory_count"`
+		ErrorMsg            string        `json:"error"`
 	}
 
 	var response Response
 	if err = json.NewDecoder(resp.Body).Decode(&response); err != nil {
-		return 0, err
+		return false, 0, err
 	}
 
-	if !response.Success {
+	if response.Success == 0 {
 		if len(response.ErrorMsg) != 0 {
-			return 0, errors.New(response.ErrorMsg)
+			return false, 0, errors.New(response.ErrorMsg)
 		}
 
-		return 0, nil // empty inventory
+		return false, 0, nil // empty inventory
 	}
 
-	var inventory map[string]*InventoryItem
-	if err = json.Unmarshal(response.Inventory, &inventory); err != nil {
-		// empty inventory...
-		// NB: This only occurs on first run...
-		return 0, nil
-	}
+	// New inventory endpoint has same order for "Assets" and "Descriptions"
+	// So we don't need create Descriptions map
+	// Now we can just incriment Descriptions array iterator with Assets iterator
+	// but when current Asset ClassID and InstanceID are equal with previos
+	// we mustn't incriment Descriptions iterator
+	//
+	// It looks like:
+	// Assets  descPos
+	//   A        0
+	//   A        0
+	//   B        1
+	//   C        2
+	//   C        2
+	//   C        2
+	//   D        3
+	descPos := 0
+	lastKey := fmt.Sprintf("%d_%d", response.Assets[0].ClassID, response.Assets[0].InstanceID)
+	for _, asset := range response.Assets {
+		tmpKey := fmt.Sprintf("%d_%d", asset.ClassID, asset.InstanceID)
+		if tmpKey != lastKey {
+			lastKey = tmpKey
+
+			descPos++
+		}
+
```

Comment:
Perhaps to fix the panic I pointed out is to check if descPos+1 < len(descriptions) and move the picture initialization here?